app/controllers/application_controller.rb

```diff
@@ -8,4 +8,12 @@ class ApplicationController < ActionController::Base
 
   before_action :authenticate_user!
   before_action :update_allowed_parameters, if: :devise_controller?
+
+  rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
+
+  private
+
+  def render_not_found
+    render 'shared/404', status: :not_found
```

Comment:
We already have `404.html` file in `/public/404.html`

Reply:
I used the same text. 
But several issues with this static html:
1. I'm not sure exactly when it will be used (only in production environment?)
2. It does not have layout
3. It does not use locals (so content would not be translated if we add more languages)

For that reason, google redirected me to this solution. 
I'd be happy to hear your solutions @junan, if  there is a way to solve point 2 and 3 with the static 404.html file ~ Thanks

Reply:
As you want to use a custom error instead of static one, then this looks fine to me